packages/medusa/src/repositories/product.ts

```diff
@@ -314,4 +323,60 @@ export class ProductRepository extends Repository<Product> {
 
     return this.findByIds(productIds)
   }
+
+  public async getFreeTextSearchResultsAndCount(
+    q: string,
+    options: CustomOptions = { where: {} },
+    relations: (keyof Product)[] = []
+  ): Promise<[Product[], number]> {
+    const cleanedOptions = this._cleanOptions(options)
+
+    let qb = this.createQueryBuilder("product")
+      .leftJoinAndSelect("product.variants", "variant")
+      .leftJoinAndSelect("product.collection", "collection")
+      .select(["product.id"])
+      .where(cleanedOptions.where)
```

Comment:
suggestion: You can move the content of the clean method here since it is pretty specific. as mention by oli this method already exists in the service :)